### PR TITLE
[BEAM-7616] apitools use urllib with the global timeout. Set it to 60 seconds # to prevent network related stuckness issues.

### PR DESCRIPTION
**Pull request by aaltay**
apitools use urllib with the global timeout. Set it to 60 seconds # to prevent network related stuckness issues.

@udim do you think this is a good solution? Do we have any other sockets, where a 60 second timeout may not be appropriate?

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/)
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)
XLang | --- | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | --- | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

File: sdks/python/apache_beam/internal/gcp/auth.py [hunk 1]
```diff
@@ -32,6 +33,10 @@
 except ImportError:
   GceAssertionCredentials = None
 
+# apitools use urllib with the global timeout. Set it to 60 seconds
+# to prevent network related stuckness issues.
+socket.setdefaulttimeout(60)
+
```
Comment (udim) [inline sdks/python/apache_beam/internal/gcp/auth.py]: The unknown part here is whether this gets applied on a remote worker. The defaulttimeout variable is a CPython implementation detail, so I don't believe it gets pickled and passed to workers. Will this module get imported on workers, or will the pickled version of it be used (such as with save_main_session)?

My notes are here:
https://issues.apache.org/jira/browse/BEAM-7616?focusedCommentId=16913549&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-16913549

Comment (aaltay) [reply sdks/python/apache_beam/internal/gcp/auth.py]: I am not sure. Adding a log statement here. I should be able to run a job and look and figure out from the logs.

File: sdks/python/apache_beam/internal/gcp/auth.py [hunk 2]
```diff
@@ -32,6 +33,10 @@
 except ImportError:
   GceAssertionCredentials = None
 
+# apitools use urllib with the global timeout. Set it to 60 seconds
+# to prevent network related stuckness issues.
+socket.setdefaulttimeout(60)
```
[the remark this answers is not in the document]
Comment (aaltay) [reply sdks/python/apache_beam/internal/gcp/auth.py]: Good point. Updated.

**Comment by aaltay** [comment]
Run Python PreCommit

**Comment by aaltay** [comment]
Run Python PostCommit

**Comment by aaltay** [comment]
Run Python 2 PostCommit

**Comment by aaltay** [comment]
I looked at the postcommit tests, I do not see the new log message. Not sure why. Save main session is not used in all post commits, so there is probably another reason. Feel free to take this over and try the other idea of applying in init.

**Comment by udim** [comment]
> I looked at the postcommit tests, I do not see the new log message. Not sure why. Save main session is not used in all post commits, so there is probably another reason. Feel free to take this over and try the other idea of applying in init.

I think that INFO logs are not printed nor saved to xunit XML output if the test is successful. Ways around this:
1. Add "--nologcapture" to `basicTestOpts` and change increase logging level to warning or error.
1. Use `print()`

**Comment by aaltay** [comment]
I should be more clear. I looked at the stackdriver logs for some of the post commit tests that ran on Dataflow. I expected worker logs to have this info statement.

**Comment by aaltay** [comment]
Run Python 2 PostCommit

**Comment by udim** [comment]
If you get no output, perhaps putting the code in `_Credentials.get_service_credentials` will do the trick (in the branch where `cls._credentials_init` is False)?

**Comment by aaltay** [comment]
Run Python 2 PostCommit

**Comment by aaltay** [comment]
Run Python 2 PostCommit

**Comment by aaltay** [comment]
Run Python PreCommit

**Comment by aaltay** [comment]
This version seems to work. I can find the new logs in stackdriver logs.

**Comment by aaltay** [comment]
Run Python 2 PostCommit

**Pull request merged**